### PR TITLE
test: Local evaluation testing

### PR DESCRIPTION
**Pull request by khvn26**
DO NOT MERGE. This contains an automated SDK test for Edge V2 migration.

To run, clone, then:

```
pytest tests/test_local_eval.py 
```

After the migration, run:

```
pytest tests/test_local_eval.py --migrated
```

### CODE REVIEW

File: tests/test_local_eval.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        spec=Flag(mock.ANY, mock.ANY, mock.ANY, mock.ANY), **mock_kwargs
+    )
+
+
+@pytest.fixture
+def environment_default_or_percentage_split_value_flag_mock() -> mock.Mock:
+    return _get_flag_mock(
+        value=get_matching_str(
+            "environment default",
+            'overridden by "split_segment"',
```
Comment (matthewelwell) [inline tests/test_local_eval.py:54]: If we can, I'd like to be more specific here. Whether the identity belongs to the percentage split segment shouldn't change before and after migration. I _think_ this is going to mean that we're just checking that it takes one of the 2 possible values.

**Review by matthewelwell — COMMENTED** [review]
I've reviewed this. It all looks ok to me and seems to check mostly the right things, the main issue (as per my comment) is that we're not checking that the values are the same before and after the migration (which I think we should expect?). 

I'd say that we should implement some way to verify this, it could just be a local json file or something, or a pickled python object written to a file maybe?

File: tests/test_local_eval.py (hunk 1 above)
Comment (khvn26) [reply tests/test_local_eval.py]: Switched to static values.

**Review by matthewelwell — APPROVED** [review]
Approving to confirm that I'm happy with the tests being performed (not as a suggestion we should merge).

**Pull request closed without merging**